### PR TITLE
Improve authentication protection

### PR DESCRIPTION
**Pull request by julian-christl**
Prevent bursts from being sent directly, but delay them to fit the requested rate.
Decrease the request rate on this endpoint

### CODE REVIEW

**Comment by Mtze** [comment]
@krusche We tested this configuration on ts1. The new configuration prevents DOS attacks on single students (e.g. Prevent a single student to login). Do yo see any problems with a stricter rate limiting on the authenticate endpoint?

**Review by Mtze — DISMISSED** [review]

File: roles/proxy/templates/nginx_artemis.conf.j2 [hunk 1]
```diff
@@ -86,7 +86,7 @@ server {
 
     location /api/authenticate {
         proxy_pass http://artemis/api/authenticate;
-        limit_req zone=loginlimit burst=5 nodelay;
+        limit_req zone=loginlimit burst=3 delay=2;
```
Comment (krusche) [inline roles/proxy/templates/nginx_artemis.conf.j2:89]: can you please add a comment above this line what it actually means?

Comment (julian-christl) [reply roles/proxy/templates/nginx_artemis.conf.j2:95]: Let me know if the comments help your understanding

Comment (krusche) [reply roles/proxy/templates/nginx_artemis.conf.j2:95]: Yes, thank you

**Comment by krusche** [comment]
> @krusche We tested this configuration on ts1. The new configuration prevents DOS attacks on single students (e.g. Prevent a single student to login). Do yo see any problems with a stricter rate limiting on the authenticate endpoint? 

Thanks, this looks good to me.

**Review by krusche — APPROVED** [review]

**Review by Mtze — APPROVED** [review]

**Pull request merged**